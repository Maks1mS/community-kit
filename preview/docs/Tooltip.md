## üìñ Detailed overview

The container component shows a text tooltip when you mouse over the content.

## ‚öôÔ∏è Usage

Add the component to the page and drop another component, for example, `Text`, inside it. Then see how it works in the preview mode.

### Auto position change

If the `Auto position change` prop is enabled, the tooltip will automatically change its position if there's not enough space. All indents will be taken into account.

## üß© Components and Props

### In the interface

| Prop name                |  Default  |   Example   |
| :----------------------- | :-------: | :---------: |
| Show Tooltip             | `Always`  | `On hover`  |
| Tooltip position         |   `Top`   |  `Bottom`   |
| Edges indent             |    `0`    |    `2px`    |
| Content indent           |   `4px`   |    `8px`    |
| Tooltip background color | `#000000` |  `#ffffff`  |
| Show the arrow           | `Checked` | `Unchecked` |
| Arrow size (px)          |   `8px`   |   `12px`    |
| Auto position change     | `Checked` | `Unchecked` |

### In the code (for developers)

| Prop name                |    Name in the code     |   Type    |  Default  |  Example  |
| :----------------------- | :---------------------: | :-------: | :-------: | :-------: |
| Show Tooltip             |   `tooltipStatusProp`   |  `enum`   | `always`  |  `hover`  |
| Tooltip position         |  `tooltipPositionProp`  |  `enum`   |   `top`   | `bottom`  |
| Edges indent             |   `tooltipOffsetProp`   | `string`  |    `0`    |   `2px`   |
| Content indent           |   `contentOffsetProp`   | `string`  |   `4px`   |   `8px`   |
| Tooltip background color |   `tooltipColorProp`    | `string`  | `#000000` | `#ffffff` |
| Show the arrow           |    `arrowStatusProp`    | `boolean` |  `true`   |  `false`  |
| Arrow size (px)          |     `arrowSizeProp`     | `string`  |   `8px`   |  `12px`   |
| Auto position change     | `tooltipAutoChangeProp` | `boolean` |  `true`   |  `false`  |

#### The 'Show Tooltip' property values

| User-friendly name | Name in the code |
| :----------------- | :--------------: |
| Always             |     `always`     |
| On hover           |     `hover`      |

#### The 'Tooltip position' property values

| User-friendly name | Name in the code |
| :----------------- | :--------------: |
| Top                |      `top`       |
| Right              |     `right`      |
| Bottom             |     `bottom`     |
| Left               |      `left`      |

## üóÑ GitHub

[Link to GitHub](https://github.com/quarkly/community-kit/tree/master/src/Tooltip)

## üóì Changelog

-   21/04/2021 (v1.0)
-   First version

## üìÆ Feedback

If you encountered a bug, please contact us so we can fix it promptly. We‚Äôre rapidly developing, so don‚Äôt hesitate to send us your feedback and request new features you would like to see added. Feel free to share what you‚Äôre working on - we **love** to see what you‚Äôre building with Quarkly!

[Help with components](https://community.quarkly.io/c/requests/11)

[We're on Discord](https://discord.gg/SuF9vCMJGW)

[Our Twitter](https://twitter.com/quarklyapp)

[dev@quarkly.io](mailto:dev@quarkly.io)

## üìù License

Licensed under the [MIT License](./LICENSE).
