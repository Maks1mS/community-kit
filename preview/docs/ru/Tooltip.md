## üìñ –ü–æ–¥—Ä–æ–±–Ω—ã–π –æ–±–∑–æ—Ä

–ö–æ–º–ø–æ–Ω–µ–Ω—Ç-–∫–æ–Ω—Ç–µ–π–Ω–µ—Ä –ø–æ–∫–∞–∑—ã–≤–∞–µ—Ç —Ç–µ–∫—Å—Ç–æ–≤—É—é –ø–æ–¥—Å–∫–∞–∑–∫—É –ø—Ä–∏ –Ω–∞–≤–µ–¥–µ–Ω–∏–∏ –∫—É—Ä—Å–æ—Ä–∞ –Ω–∞ —Å–æ–¥–µ—Ä–∂–∏–º–æ–µ.

## ‚öôÔ∏è –ò—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏–µ

–î–æ–±–∞–≤—å—Ç–µ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç –Ω–∞ —Å—Ç—Ä–∞–Ω–∏—Ü—É –∏ –ø–æ–º–µ—Å—Ç–∏—Ç–µ –≤ –Ω–µ–≥–æ –ª—é–±–æ–π –¥—Ä—É–≥–æ–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç, –Ω–∞–ø—Ä–∏–º–µ—Ä, `Text`. –ó–∞—Ç–µ–º, –ø–æ—Å–º–æ—Ç—Ä–∏—Ç–µ –∫–∞–∫ –æ–Ω —Ä–∞–±–æ—Ç–∞–µ—Ç –≤ —Ä–µ–∂–∏–º–µ –ø—Ä–µ–≤—å—é.

### –ê–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∞—è —Å–º–µ–Ω–∞ –ø–æ–ª–æ–∂–µ–Ω–∏—è

–ï—Å–ª–∏ —Å–≤–æ–π—Å—Ç–≤–æ `–ê–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∞—è —Å–º–µ–Ω–∞ –ø–æ–ª–æ–∂–µ–Ω–∏—è` –∞–∫—Ç–∏–≤–Ω–æ, –≤—Å–ø–ª—ã–≤–∞—é—â–∞—è –ø–æ–¥—Å–∫–∞–∑–∫–∞ –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ –±—É–¥–µ—Ç –º–µ–Ω—è—Ç—å —Å–≤–æ–µ –ø–æ–ª–æ–∂–µ–Ω–∏–µ –≤ —Å–ª—É—á–∞–µ –Ω–µ—Ö–≤–∞—Ç–∫–∏ –º–µ—Å—Ç–∞. –ü—Ä–∏ —ç—Ç–æ–º –±—É–¥—É—Ç —É—á–∏—Ç—ã–≤–∞—Ç—å—Å—è –≤—Å–µ –∏–º–µ—é—â–∏–µ—Å—è –æ—Ç—Å—Ç—É–ø—ã.

## üß© –ö–æ–º–ø–æ–Ω–µ–Ω—Ç—ã –∏ —Å–≤–æ–π—Å—Ç–≤–∞

### –í –∏–Ω—Ç–µ—Ä—Ñ–µ–π—Å–µ

| –ù–∞–∑–≤–∞–Ω–∏—è —Å–≤–æ–π—Å—Ç–≤               | –ü–æ —É–º–æ–ª—á–∞–Ω–∏—é |    –ü—Ä–∏–º–µ—Ä    |
| :----------------------------- | :----------: | :----------: |
| –ü–æ–∫–∞–∑—ã–≤–∞—Ç—å Tooltip             |   `–í—Å–µ–≥–¥–∞`   |   `–í—Å–µ–≥–¥–∞`   |
| –ü–æ–ª–æ–∂–µ–Ω–∏–µ Tooltip              |   `–°–≤–µ—Ä—Ö—É`   |   `–°–Ω–∏–∑—É`    |
| –û—Ç—Å—Ç—É–ø –æ—Ç –∫—Ä–∞–µ–≤                |     `0`      |    `2px`     |
| –û—Ç—Å—Ç—É–ø –æ—Ç –∫–æ–Ω—Ç–µ–Ω—Ç–∞             |    `4px`     |    `8px`     |
| –¶–≤–µ—Ç —Ñ–æ–Ω–∞ Tooltip              |  `#000000`   |  `#ffffff`   |
| –ü–æ–∫–∞–∑–∞—Ç—å —Å—Ç—Ä–µ–ª–æ—á–∫—É             |  `–û—Ç–º–µ—á–µ–Ω`   | `–ù–µ –æ—Ç–º–µ—á–µ–Ω` |
| –†–∞–∑–º–µ—Ä —Å—Ç—Ä–µ–ª–æ—á–∫–∏ (px)          |    `8px`     |    `12px`    |
| –ê–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∞—è —Å–º–µ–Ω–∞ –ø–æ–ª–æ–∂–µ–Ω–∏—è |  `–û—Ç–º–µ—á–µ–Ω`   | `–ù–µ –æ—Ç–º–µ—á–µ–Ω` |

### –í –∫–æ–¥–µ (–¥–ª—è —Ä–∞–∑—Ä–∞–±–æ—Ç—á–∏–∫–æ–≤)

| –ù–∞–∑–≤–∞–Ω–∏—è —Å–≤–æ–π—Å—Ç–≤               |     –ù–∞–∑–≤–∞–Ω–∏–µ –≤ –∫–æ–¥–µ     |    –¢–∏–ø    | –ü–æ —É–º–æ–ª—á–∞–Ω–∏—é |  –ü—Ä–∏–º–µ—Ä   |
| :----------------------------- | :---------------------: | :-------: | :----------: | :-------: |
| –ü–æ–∫–∞–∑—ã–≤–∞—Ç—å Tooltip             |   `tooltipStatusProp`   |  `enum`   |   `always`   |  `hover`  |
| –ü–æ–ª–æ–∂–µ–Ω–∏–µ Tooltip              |  `tooltipPositionProp`  |  `enum`   |    `top`     | `bottom`  |
| –û—Ç—Å—Ç—É–ø –æ—Ç –∫—Ä–∞–µ–≤                |   `tooltipOffsetProp`   | `string`  |     `0`      |   `2px`   |
| –û—Ç—Å—Ç—É–ø –æ—Ç –∫–æ–Ω—Ç–µ–Ω—Ç–∞             |   `contentOffsetProp`   | `string`  |    `4px`     |   `8px`   |
| –¶–≤–µ—Ç —Ñ–æ–Ω–∞ Tooltip              |   `tooltipColorProp`    | `string`  |  `#000000`   | `#ffffff` |
| –ü–æ–∫–∞–∑–∞—Ç—å —Å—Ç—Ä–µ–ª–æ—á–∫—É             |    `arrowStatusProp`    | `boolean` |    `true`    |  `false`  |
| –†–∞–∑–º–µ—Ä —Å—Ç—Ä–µ–ª–æ—á–∫–∏ (px)          |     `arrowSizeProp`     | `string`  |    `8px`     |  `12px`   |
| –ê–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∞—è —Å–º–µ–Ω–∞ –ø–æ–ª–æ–∂–µ–Ω–∏—è | `tooltipAutoChangeProp` | `boolean` |    `true`    |  `false`  |

#### –ó–Ω–∞—á–µ–Ω–∏—è —Å–≤–æ–π—Å—Ç–≤–∞ '–ü–æ–∫–∞–∑—ã–≤–∞—Ç—å Tooltip'

| –ü–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª—å—Å–∫–æ–µ –Ω–∞–∑–≤–∞–Ω–∏–µ | –ù–∞–∑–≤–∞–Ω–∏–µ –≤ –∫–æ–¥–µ |
| :------------------------ | :-------------: |
| –í—Å–µ–≥–¥–∞                    |    `always`     |
| –ü—Ä–∏ –Ω–∞–≤–µ–¥–µ–Ω–∏–∏             |     `hover`     |

#### –ó–Ω–∞—á–µ–Ω–∏—è —Å–≤–æ–π—Å—Ç–≤–∞ '–ü–æ–ª–æ–∂–µ–Ω–∏–µ Tooltip'

| –ü–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª—å—Å–∫–æ–µ –Ω–∞–∑–≤–∞–Ω–∏–µ | –ù–∞–∑–≤–∞–Ω–∏–µ –≤ –∫–æ–¥–µ |
| :------------------------ | :-------------: |
| –°–≤–µ—Ä—Ö—É                    |      `top`      |
| –°–ø—Ä–∞–≤–∞                    |     `right`     |
| –°–Ω–∏–∑—É                     |    `bottom`     |
| –°–ª–µ–≤–∞                     |     `left`      |

## üóÑ GitHub

[–°—Å—ã–ª–∫–∞ –Ω–∞ GitHub](https://github.com/quarkly/community-kit/tree/master/src/Tooltip)

## üóì Changelog

-   21/04/2021 (v1.0)
-   –ü–µ—Ä–≤–∞—è –≤–µ—Ä—Å–∏—è

## üìÆ Feedback

If you encountered a bug, please contact us so we can fix it promptly. We‚Äôre rapidly developing, so don‚Äôt hesitate to send us your feedback and request new features you would like to see added. Feel free to share what you‚Äôre working on - we **love** to see what you‚Äôre building with Quarkly!

[Help with components](https://community.quarkly.io/c/requests/11)

[We're on Discord](https://discord.gg/f9KhSMGX)

[Our Twitter](https://twitter.com/quarklyapp)

[dev@quarkly.io](mailto:dev@quarkly.io)

## üìù License

Licensed under the [MIT License](https://raw.githubusercontent.com/quarkly/community-kit/master/LICENSE).
