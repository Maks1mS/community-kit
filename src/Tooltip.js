import React from 'react';
import atomize from "@quarkly/atomize";

const Tooltip = props => <div {...props}>Say hello Tooltip</div>

export default atomize(Tooltip)({
  name: "Tooltip",
  effects: {
    hover: ":hover"
  },
  description: {
    // paste here description for your component
    en:
      "Tooltip â€” my awesome component",
  },
  propInfo: {
    // paste here props description for your component
    yourCustomProps: {
      control: "input"
    }
  }
});
