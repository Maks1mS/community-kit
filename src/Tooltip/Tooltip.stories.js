import React from 'react';
import { Text } from '@quarkly/widgets';
import Tooltip from './Tooltip';
import { argTypes } from '../modules';

export default {
    title: 'Tooltip',
    component: Tooltip,
    args: Tooltip.defaultProps,
    argTypes: argTypes(Tooltip.propInfo, Tooltip.defaultProps),
};

export const StoryDefault = (props) => (
    <Tooltip {...props}>
        <Text>{'Some text'}</Text>
    </Tooltip>
);

export const TooltipPositions = (props) => (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            padding: '100px',
        }}
    >
        <Tooltip {...props}>
            <Text>{'Some text'}</Text>
        </Tooltip>
    </div>
);
