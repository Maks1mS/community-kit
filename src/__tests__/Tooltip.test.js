import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from '../Tooltip';

describe('Tooltip component', () => {
    it('renders correctly without params', () => {
        const component = renderer.create(<Tooltip />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
