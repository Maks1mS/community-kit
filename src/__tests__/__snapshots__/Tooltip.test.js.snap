// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly without params 1`] = `
<div
  className="sc-eCImvq gIpXVq"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="sc-eCImvq bbiifI"
    display="flex"
    height="100%"
    width="100%"
  >
    <div
      className="sc-eCImvq dBvtEa"
      data-qkind="Box"
      data-qoverride="Wrapper"
      display="flex"
      opacity={1}
      visibility="visible "
      width="max-content"
    >
      <div
        className="sc-dkPtyc gvLftD"
        color="white"
        data-qkind="Text"
        data-qoverride="Title"
      >
        Some text
      </div>
      <div
        className="sc-eCImvq bTukaB"
        data-qkind="Box"
        data-qoverride="Arrow"
        opacity={1}
        visibility="visible "
      />
    </div>
  </div>
  <div
    className="sc-eCImvq kDnecC"
    data-qkind="Box"
    data-qoverride="Content"
    width="100%"
  />
</div>
`;
